app/jobs/idd2wdl.py

```diff
@@ -138,16 +149,32 @@ def target_filename(target, index):
     step_status_lock=contextlib.suppress()
   )
   step_instance.input_files_local = {input_files_local}
+
   with open(step_instance.step_status_local, "w") as status_file:
     json.dump(dict(), status_file)
+
   try:
     {count_input_reads}
     step_instance.update_status_json_file("running")
     step_instance.run()
     step_instance.count_reads()
     step_instance.save_counts()
-    step_instance.update_status_json_file("finished_running")
-  except Exception as e:
+    # temporary until we instrument miniwdl - not yet uploaded, but this is the final status
+    step_instance.update_status_json_file("uploaded")
+  except:
+    # process exception for status reporting
+    try:
+      raise
+    except idseq_dag.engine.pipeline_step.InvalidInputFileError as e:
+      try:
+        self.update_status_json_file("user_errored")
+      except:
+        logging.error("Failed to update status to 'user_errored'")
+    except:
+      try:
+        self.update_status_json_file("pipeline_errored")
+      except:
+        logging.error("Failed to update status to 'pipeline_errored'")
     traceback.print_exc()
     exit(json.dumps(dict(wdl_error_message=True, error=type(e).__name__, cause=str(e))))
```

Comment:
With the change above, `e` will now end up unbound in the "pipeline_errored" branch above. If you just keep L164 as ` except Exception as e:` this will work, or consider rewriting it like this instead:
```python
except Exception as e:
    status = "user_errored" if isinstance(e, idseq_dag.engine.pipeline_step.InvalidInputFileError) else "pipeline_errored"
    try:
        self.update_status_json_file(status)
    except:
        logging.error("Failed to update status to '%s'", status)
    traceback.print_exc()
    exit(json.dumps(dict(wdl_error_message=True, error=type(e).__name__, cause=str(e))))
```

Reply:
Sure, makes sense and it is more compact.